src/main/java/core/basesyntax/service/impl/ReaderServiceImpl.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.ReaderService;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class ReaderServiceImpl implements ReaderService {
+    @Override
+    public List<String> readFile(String filePath) {
+        Scanner scanner;
+        try {
+            scanner = new Scanner(new File(filePath));
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+        ArrayList<String> list = new ArrayList<>();
+        while (scanner.hasNext()) {
+            list.add(scanner.next());
+        }
+        scanner.close();
```

Comment:
Use try-with-resources for the Scanner to ensure it is closed properly even in case of an exception. This avoids the need to explicitly close the scanner and is more robust.